tests/arch/x86/info/src/acpi.c

```diff
@@ -4,145 +4,47 @@
-	for (; n_path > 0; n_path--) {
-		printk("\t\t\t. Path D:%u F:%u\n",
-		       path->device, path->function);
-		path = (struct acpi_dmar_dev_path *)(POINTER_TO_UINT(path) +
-						     ACPI_DMAR_DEV_PATH_SIZE);
-	}
-
-	printk("\n");
+	printk("IOAPIC Info: Bus: %d, dev:%d, fun:%d\n", dmar_id->bits.bus, dmar_id->bits.device,
+	       dmar_id->bits.function);
 }
 
-static void vtd_drhd_info(struct acpi_drhd *drhd)
-{
-	struct acpi_dmar_dev_scope *dev_scope;
-	int n_ds, i;
-
-	if (drhd->flags & ACPI_DRHD_FLAG_INCLUDE_PCI_ALL) {
-		printk("\t\t- Includes all PCI devices");
-	} else {
-		printk("\t\t- Includes only listed PCI devices");
-	}
-
-	printk(" under given Segment\n");
-
-	printk("\t\t- Segment number %u\n", drhd->segment_num);
-	printk("\t\t- Base Address 0x%llx\n", drhd->base_address);
-
-	dev_scope = z_acpi_get_drhd_dev_scopes(drhd, &n_ds);
-	if (dev_scope == NULL) {
-		printk("\t\t- No device scopes\n");
-		return;
-	}
-
-	printk("\t\t- Device Scopes:\n");
-	for (i = 0; i < n_ds; i++) {
-		vtd_dev_scope_info(dev_scope);
-		dev_scope = (struct acpi_dmar_dev_scope *)(
-			POINTER_TO_UINT(dev_scope) + dev_scope->length);
-	}
-
-	printk("\n");
-}
-
-static void vtd_info(void)
+void acpi(void)
 {
-	struct acpi_dmar *dmar;
+	int nr_cpus = 0, i, inst_cnt;
+	struct acpi_madt_local_apic *lapic[CONFIG_MP_MAX_NUM_CPUS];
 
-	dmar = z_acpi_find_dmar();
-	if (dmar == NULL) {
-		printk("\tIntel VT-D not supported or exposed\n");
+	if (acpi_madt_entry_get(ACPI_MADT_TYPE_LOCAL_APIC, (ACPI_SUBTABLE_HEADER **)lapic,
+		&inst_cnt)) {
+		printk("error on get MAD table\n");
 		return;
 	}
 
-	printk("\tIntel VT-D Supported:\n");
-
-	printk("\t-> X2APIC ");
-	if (dmar->flags & ACPI_DMAR_FLAG_X2APIC_OPT_OUT) {
-		printk("should be opted out\n");
-	} else {
-		printk("does not need to be opted out\n");
-	}
-
-	if (dmar->flags & ACPI_DMAR_FLAG_INTR_REMAP) {
-		struct acpi_drhd *drhd;
-		int hw_n, i;
-
-		printk("\t-> Interrupt remapping supported\n");
-
-		drhd = z_acpi_find_drhds(&hw_n);
-		printk("\t-> %u remapping hardware found:\n", hw_n);
-
-		for (i = 0; i < hw_n; i++) {
-			printk("\t\tDRHD %u:\n", i);
-			vtd_drhd_info(drhd);
-			drhd = (struct acpi_drhd *)(POINTER_TO_UINT(drhd) +
-						    drhd->entry.length);
+	/* count number of CPUs present which are enabled*/
+	for (i = 0; i < CONFIG_MP_MAX_NUM_CPUS; i++) {
+		if (lapic[i]->LapicFlags & 1u) {
+			nr_cpus++;
 		}
-	} else {
-		printk("\t-> Interrupt remapping not supported\n");
-	}
-}
-
-void acpi(void)
-{
-	int nr_cpus;
-
-	for (nr_cpus = 0; z_acpi_get_cpu(nr_cpus); ++nr_cpus) {
-		/* count number of CPUs present */
 	}
 
 	if (nr_cpus == 0) {
 		printk("ACPI: no RSDT/MADT found\n\n");
 	} else {
 		printk("ACPI: %d CPUs found\n", nr_cpus);
 
-		for (int i = 0; i < nr_cpus; ++i) {
-			struct acpi_cpu *cpu = z_acpi_get_cpu(i);
-			printk("\tCPU #%d: APIC ID 0x%02x\n", i, cpu->apic_id);
+		for (i = 0; i < CONFIG_MP_MAX_NUM_CPUS; ++i) {
+			printk("\tCPU #%d: APIC ID 0x%02x\n", i, lapic[i]->Id);
```

Comment:
No this is bogus: it can happen that a defined CPU in madt might not be enabled at all,  so we don't want to display those. Thus the logic above in acpi() that first counts the valid CPUs. Don't change that.

Reply:
Nevermind this comment, I forgot that z_acpi_get_cpu() does the check for disablement. 

and yeah, the above logic is used only to verify that we got rsdt/madt.